mne/time_frequency/tests/test_csd.py

```diff
@@ -161,3 +161,161 @@ def test_compute_epochs_csd_on_artificial_data():
+    power = np.mean(tfr.data, 2)
+
+    # Maximum PSD should occur for specific channel
+    max_ch_power = power.argmax()
+    max_ch_mt = data_csd_mt.diagonal().argmax()
+    max_ch_fourier = data_csd_fourier.diagonal().argmax()
+    assert_equal(max_ch_mt, max_ch_power)
+    assert_equal(max_ch_fourier, max_ch_power)
+
+    # Maximum CSD should occur for specific channel
+    ch_csd_mt = np.abs(data_csd_mt[max_ch_power])
+    ch_csd_mt[max_ch_power] = 0.
+    max_ch_csd_mt = np.argmax(ch_csd_mt)
+    ch_csd_fourier = np.abs(data_csd_fourier[max_ch_power])
+    ch_csd_fourier[max_ch_power] = 0.
+    max_ch_csd_fourier = np.argmax(ch_csd_fourier)
+    assert_equal(max_ch_csd_mt, max_ch_csd_fourier)
+
+    # Check a list of CSD matrices is returned for multiple frequencies within
+    # a given range when fsum=False
+    csd_fsum, freqs_fsum = csd_array(X, sfreq, mode='fourier', fmin=8,
+                                     fmax=20, fsum=True)
+    csds, freqs = csd_array(X, sfreq, mode='fourier', fmin=8, fmax=20,
+                            fsum=False)
+
+    csd_sum = np.sum(csds, axis=2)
+
+    assert_equal(csds.shape[2], 2)
+    assert_equal(len(freqs), 2)
+    assert_array_equal(freqs_fsum, freqs)
+    assert_array_equal(csd_fsum, csd_sum)
+
+
+def test_csd_on_artificial_data():
+    """Test computing CSD on artificial data. """
+    epochs = _get_data(mode='sin')
+    sfreq = epochs.info['sfreq']
+
+    # Computing signal power in the time domain
+    signal_power = sum_squared(epochs._data)
+    signal_power_per_sample = signal_power / len(epochs.times)
+
+    # Computing signal power in the frequency domain
+    data_csd_mt, freqs_mt = csd_array(epochs._data, sfreq,
+                                      mode='multitaper')
+    data_csd_fourier, freqs_fft = csd_array(epochs._data, sfreq,
+                                            mode='fourier')
+
+    fourier_power = np.abs(data_csd_fourier[0, 0]) * sfreq
+    mt_power = np.abs(data_csd_mt[0, 0]) * sfreq
+    assert_true(abs(fourier_power - signal_power) <= 0.5)
+    assert_true(abs(mt_power - signal_power) <= 1)
+
+    # Power per sample should not depend on time window length
+    for tmax in [0.2, 0.8]:
+        tend = np.where(epochs.times <= tmax)[0][-1] + 1
+        tslice = slice(None, tend, None)
+
+        for add_n_fft in [0, 30]:
+            t_mask = (epochs.times >= 0) & (epochs.times <= tmax)
+            n_samples = sum(t_mask)
+            n_fft = n_samples + add_n_fft
+
+            data_csd_fourier, _ = csd_array(epochs._data[:, :, tslice],
+                                            sfreq, mode='fourier',
+                                            fmin=0, fmax=np.inf, n_fft=n_fft)
+
+            fourier_power_per_sample = np.abs(data_csd_fourier[0, 0]) *\
+                sfreq / n_fft
+            assert_true(abs(signal_power_per_sample -
+                            fourier_power_per_sample) < 0.003)
```

Comment:
@kingjr This is what was present in the existing tests that I adapted for the new functions. I think questioning the `0.003` belongs in a separate issue since it was already present in the test prior to my edits.